yb-voyager/src/utils/gcs/gcsutils.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright (c) YugaByte, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package gcs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+var client *storage.Client
+
+func createClientIfNotExists() {
+	if client != nil {
+		return
+	}
+	// Creates a client with default credentials.
+	c, err := storage.NewClient(context.Background())
+	if err != nil {
+		utils.ErrExit("create gcs client: %w", err)
+	}
+	client = c
+
+}
+
+func ValidateObjectURL(datadir string) error {
+	u, err := url.Parse(datadir)
+	if err != nil {
+		return err
+	}
+	bucket := u.Host
+	if bucket == "" {
+		return fmt.Errorf("missing bucket in gcs url %v", datadir)
+	}
+	return nil
+}
+
+func SplitObjectPath(objectPath string) (string, string, error) {
+	u, err := url.Parse(objectPath)
+	if err != nil {
+		return "", "", err
+	}
+	bucket := u.Host
+	key := u.Path[1:] //remove initial "/", unable to find object with it
+	if bucket == "" {
+		return "", "", fmt.Errorf("missing bucket in gcs url %v", objectPath)
+	}
+	if key == "" {
+		return "", "", fmt.Errorf("missing key in gcs url %v", objectPath)
+	}
+	return bucket, key, nil
+}
+
+func ListAllObjects(bucket string) ([]string, error) {
```

Comment:
List all objects from a bucket might be too costly depending on the size of the bucket. We should arrange the code so that **at most** we fetch all objects under the `--data-dir`. Remember, the URL given in the `--data-dir` can include bucket name as well as prefix/key.